### PR TITLE
Header / Footer / Home / Artists / Concerts / Podcasts in progress local API

### PR DESCRIPTION
**Pull request by H4lfbeard**
We created header / footer components and display blocks and image header components. The display blocks components fetch the local API to display Artists or Playlists. The searchbar in the header is not functional yet.

### CODE REVIEW

File: client/src/App.css [hunk 1]
```diff
@@ -2,147 +2,14 @@
   margin: 0;
   padding: 0;
   box-sizing: border-box;
+  font-family: "Open sans", sans-serif;
```
Comment (Yoann-Fortin) [inline client/src/App.css]: Use your variable please

File: client/src/components/Footer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+function Footer() {
+  return (
+    <>
+      <footer>
+        <a href="#top">
+          <p>Retour en haut de page</p>
+        </a>
+        <p className="copyright">
+          ©copyright 2024 <span className="logo">WILDIFY</span>
```
Comment (Yoann-Fortin) [inline client/src/components/Footer/index.tsx]: Use technical CSS please to uppercase

File: client/src/components/Header/Navbar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+function Navbar({ showLinks, setShowLinks }: NavbarProps) {
+  const handleShowLinks = () => {
+    setShowLinks(!showLinks);
+  };
+  return (
+    <>
+      <Link to="/" className="logo-link">
+        <p className="logo-text">WILDIFY</p>
```
Comment (Yoann-Fortin) [inline client/src/components/Header/Navbar/index.tsx]: Is wrong on semantic

File: client/src/components/Header/Navbar/style.css [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+.show-nav {
+  right: 0vw;
+  width: 100vw;
+  visibility: visible;
+}
+
+.navbar-burger {
+  display: block;
+  top: 1rem;
```
Comment (Yoann-Fortin) [inline client/src/components/Header/Navbar/style.css]: Dead code ?

File: client/src/components/Header/Pseudo/index.tsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import "./style.css";
+
+function Pseudo() {
+  return (
+    <aside className="pseudo">
```
Comment (Yoann-Fortin) [inline client/src/components/Header/Pseudo/index.tsx:5]: Is not pertinent here

File: client/src/components/Header/Searchbar/index.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import "./style.css";
+
+function Searchbar() {
+  return (
+    <form action="">
+      <input type="search" placeholder="Recherche..." value="" />
```
Comment (Yoann-Fortin) [inline client/src/components/Header/Searchbar/index.tsx]: label or aria-label is necessary here

File: client/src/components/Header/style.css [hunk 7]
```diff
@@ -0,0 +1,18 @@
+header {
+  max-height: 20rem;
+  background-color: var(--bgColor);
+  width: 100%;
+  font-weight: bold;
+  display: grid;
+  grid-template-columns: repeat(3, 1fr);
+  grid-template-rows: repeat(3, 1fr);
+  padding: 1rem 0 1rem 0;
```
Comment (Yoann-Fortin) [inline client/src/components/Header/style.css]: This declaration will be reduce on 1rem 0

File: client/src/components/displayBlocks/MyArtistsSearch.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  };
+  images: {
+    url: string;
+  }[];
+};
+
+export default function ArtistSearch() {
+  const [userArtists, setUserArtists] = useState<ArtistItem[]>([]);
+
+  useEffect(() => {
```
Comment (Yoann-Fortin) [inline client/src/components/displayBlocks/MyArtistsSearch.tsx:18]: Try to use loader with react router dom

File: client/src/components/displayBlocks/MyBlocks.tsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  }[];
+};
+
+type MyBlocksProps = {
+  items: DisplayItem[];
+};
+
+export default function MyBlocks({ items }: MyBlocksProps) {
+  return (
+    <>
```
Comment (Yoann-Fortin) [inline client/src/components/displayBlocks/MyBlocks.tsx]: Dead code

File: client/src/components/displayBlocks/MyPlaylistSearch.tsx [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  };
+  images: {
+    url: string;
+  }[];
+};
+
+export default function GetPlaylists() {
+  const [userPlaylists, setUserPlaylists] = useState<PlaylistItem[]>([]);
+
+  useEffect(() => {
```
Comment (Yoann-Fortin) [inline client/src/components/displayBlocks/MyPlaylistSearch.tsx:18]: See comment above

File: client/src/components/displayBlocks/MyPodcastsSearch.tsx
Comment (Yoann-Fortin) [inline client/src/components/displayBlocks/MyPodcastsSearch.tsx:1]: No comment

File: client/src/components/imageBlock/imageBlockArtists.tsx
Comment (Yoann-Fortin) [inline client/src/components/imageBlock/imageBlockArtists.tsx:1]: More duplication, extract this please

File: client/src/components/imageBlock/style.css [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+  position: absolute;
+  margin: 30% 0 0 10%;
+  font-size: 5rem;
+  color: var(--textColorWhite);
+  font-family: var(--titleFont);
+}
+
+@media (max-width: 768px) {
+  section.background-image h1 {
+    position: absolute;
```
Comment (Yoann-Fortin) [inline client/src/components/imageBlock/style.css]: Dead code

File: client/src/pages/Artists-Album/ArtistList.tsx [hunk 12]
```diff
@@ -13,13 +13,13 @@ function ArtistList({ artists }: ArtistListProps) {
   return (
     <div className="ArtistList">
```
Comment (Yoann-Fortin) [inline client/src/pages/Artists-Album/ArtistList.tsx]: No comment

File: client/src/pages/Artists-Album/ArtistList.tsx [hunk 13]
```diff
@@ -13,13 +13,13 @@ function ArtistList({ artists }: ArtistListProps) {
...
       {artists.map((artist) => (
-        <figure key={artist.id} className="DisplayEmployee">
+        <figure key={artist.id} className="Artists">
           <img src={artist.picture} alt={artist.name} />
           <figcaption>
             <strong>{artist.name}</strong>
             {artist.genres && artist.genres.length > 0 && (
               <div className="genres">
```
Comment (Yoann-Fortin) [inline client/src/pages/Artists-Album/ArtistList.tsx]: No comment

File: client/src/pages/Artists-Album/ArtistSearch.tsx [hunk 14]
```diff
@@ -1,24 +1,24 @@
...
   const getArtists = () => {
-    fetch("http://localhost:3310/api/artists")
+    fetch(`${import.meta.env.VITE_API_URL}/api/artists`)
       .then((response) => response.json())
       .then((data) => {
         setArtists(data.results);
       });
   };
   return (
     <div className="App">
```
Comment (Yoann-Fortin) [inline client/src/pages/Artists-Album/ArtistSearch.tsx]: same

File: client/src/pages/Artists-Album/Artists.tsx [hunk 15]
```diff
@@ -0,0 +1,16 @@
+import ArtistSearch from "./ArtistSearch";
+// import "./Artists.css";
+
+export default function Artist() {
+  return (
+    <>
+      <img
+        className="Picture"
```
Comment (Yoann-Fortin) [inline client/src/pages/Artists-Album/Artists.tsx]: camelcase

File: server/database/schema.sql
Comment (Yoann-Fortin) [inline server/database/schema.sql:1]: No possible

**Review by Yoann-Fortin — CHANGES_REQUESTED** [review]

File: client/src/App.css (hunk 1 above)
Comment (H4lfbeard) [reply client/src/App.css]: done

File: client/src/components/Footer/index.tsx (hunk 2 above)
Comment (H4lfbeard) [reply client/src/components/Footer/index.tsx]: done

File: client/src/components/displayBlocks/MyBlocks.tsx (hunk 9 above)
Comment (H4lfbeard) [reply client/src/components/displayBlocks/MyBlocks.tsx]: done

File: client/src/components/displayBlocks/MyPodcastsSearch.tsx
Comment (H4lfbeard) [reply client/src/components/displayBlocks/MyPodcastsSearch.tsx:1]: done

File: client/src/components/imageBlock/style.css (hunk 11 above)
Comment (H4lfbeard) [reply client/src/components/imageBlock/style.css]: done

File: client/src/pages/Artists-Album/ArtistList.tsx (hunk 12 above)
Comment (H4lfbeard) [reply client/src/pages/Artists-Album/ArtistList.tsx]: deleted

File: client/src/pages/Artists-Album/ArtistList.tsx (hunk 13 above)
Comment (H4lfbeard) [reply client/src/pages/Artists-Album/ArtistList.tsx]: deleted

File: client/src/pages/Artists-Album/ArtistSearch.tsx (hunk 14 above)
Comment (H4lfbeard) [reply client/src/pages/Artists-Album/ArtistSearch.tsx]: deleted

File: client/src/pages/Artists-Album/Artists.tsx (hunk 15 above)
Comment (H4lfbeard) [reply client/src/pages/Artists-Album/Artists.tsx]: deleted

File: client/src/components/Header/Navbar/index.tsx (hunk 3 above)
Comment (Ccilpy) [reply client/src/components/Header/Navbar/index.tsx]: done, p was delete

File: client/src/components/Header/Navbar/style.css (hunk 4 above)
Comment (Ccilpy) [reply client/src/components/Header/Navbar/style.css]: done, code deleted

File: client/src/components/Header/Searchbar/index.tsx (hunk 6 above)
Comment (Ccilpy) [reply client/src/components/Header/Searchbar/index.tsx]: done

File: client/src/components/Header/style.css (hunk 7 above)
Comment (Ccilpy) [reply client/src/components/Header/style.css]: done

File: client/src/components/Header/Navbar/index.tsx [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+function Navbar({ showLinks, setShowLinks }: NavbarProps) {
+  const handleShowLinks = () => {
+    setShowLinks(!showLinks);
+  };
+  return (
+    <>
+      <Link to="/" className="logo-link">
+        WILDIFY
```
Comment (Yoann-Fortin) [inline client/src/components/Header/Navbar/index.tsx:16]: To uppercase will be better

File: client/src/components/displayBlocks/MyBlocks.tsx [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+              >
+                {item.images[0] && (
+                  <img src={item.images[0].url} alt={item.name} />
+                )}
+                <h3>{item.name}</h3>
+              </a>
+            </li>
+          ))
+        ) : (
+          <p>Mes données YOvont s'afficher ici</p>
```
Comment (Yoann-Fortin) [inline client/src/components/displayBlocks/MyBlocks.tsx:38]: Is "YOvont" on purpose ?

File: client/src/components/displayBlocks/MyPodcastsSearch.tsx [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+  images: {
+    url: string;
+  }[];
+};
+
+export default function PodcastSearch() {
+  const [userPodcasts, setUserPodcasts] = useState<PodcastItem[]>([]);
+
+  useEffect(() => {
+    fetch(`${import.meta.env.VITE_API_URL}/api/userPodcasts`)
```
Comment (Yoann-Fortin) [inline client/src/components/displayBlocks/MyPodcastsSearch.tsx:19]: Try to extract this on loader

File: server/database/schema.sql
Comment (Yoann-Fortin) [inline server/database/schema.sql:1]: Why this file ?

**Pull request merged**